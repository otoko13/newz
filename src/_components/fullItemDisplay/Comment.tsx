import React from 'react';
import moment from 'moment';
import './comment.scss';	
import { INewsItem } from '../newsService';
import CommentsSection from './CommentsSection';

export interface ICommentProps {	
    comment: INewsItem;	
}	

function getFormattedDate(time: number) {
    return moment.utc(time).format('HH:mm, dddd MMMM Do YYYY');
}

const Comment = (props: ICommentProps) => 
    <div className='Comment'>
        <div className='by'>{props.comment.by}</div>
        <div className='date'>{getFormattedDate(props.comment.time * 1000)}</div>
        <div className='comment-content' dangerouslySetInnerHTML={{__html: props.comment.text || ''}} />
        {(props.comment.kids && props.comment.kids.length > 0) &&
            <CommentsSection newsItem={props.comment} />
        }
    </div>;

export default Comment; 