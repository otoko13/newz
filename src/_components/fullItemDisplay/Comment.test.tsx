import { render } from '@testing-library/react';
import * as React from 'react';
import Comment, { ICommentProps } from './Comment';

let props: ICommentProps;

beforeEach(() => {
    props = {
        comment: {
            id: 1,
            descendants: 0,
            by: 'King Julian',
            kids: [],
            time: 1581442689,
            title: 'A comment',
            text: 'I believe',
            type: 'story',
        }
    }
});

it('should look sensible', () => {
    const { container } = render(<Comment {...props} />);
    expect(container.firstChild).toMatchSnapshot();
});