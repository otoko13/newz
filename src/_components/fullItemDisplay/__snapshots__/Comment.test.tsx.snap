// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should look sensible 1`] = `
<div
  class="Comment"
>
  <div
    class="by"
  >
    King Julian
  </div>
  <div
    class="date"
  >
    17:38, Tuesday February 11th 2020
  </div>
  <div
    class="comment-content"
  >
    I believe
  </div>
</div>
`;
